// @mintlify/validation/dist/mint-config/upgrades/upgradeToDocsConfig.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import _ from 'lodash';
import { updateNavigationToDocsConfig } from './updateNavigationToDocsConfig.js';
function updateRounded(config) {
    return config.rounded === 'sharp' ? 'sharp' : 'regular';
}
function updateEyebrows(config) {
    var _a;
    return ((_a = config.eyebrow) === null || _a === void 0 ? void 0 : _a.display) === 'section' ? 'section' : 'breadcrumbs';
}
function updateCodeblocks(config) {
    var _a;
    return ((_a = config.codeBlock) === null || _a === void 0 ? void 0 : _a.mode) === 'dark' ? 'dark' : 'system';
}
function updateApiPlayground(config) {
    var _a, _b, _c, _d;
    const oldMode = (_b = (_a = config.api) === null || _a === void 0 ? void 0 : _a.playground) === null || _b === void 0 ? void 0 : _b.mode;
    const disableProxy = (_d = (_c = config.api) === null || _c === void 0 ? void 0 : _c.playground) === null || _d === void 0 ? void 0 : _d.disableProxy;
    if (!oldMode && disableProxy === undefined)
        return undefined;
    const display = (() => {
        switch (oldMode) {
            case 'show':
                return 'interactive';
            case 'hide':
            case undefined:
                return 'none';
            default:
                return oldMode;
        }
    })();
    const proxy = disableProxy !== undefined ? !disableProxy : undefined;
    return { display, proxy };
}
function updateTopbarPrimary(config) {
    const ctaButton = config.topbarCtaButton;
    if (!ctaButton)
        return undefined;
    if (ctaButton.type === undefined || ctaButton.type === 'link') {
        return {
            type: 'button',
            label: ctaButton.name,
            href: ctaButton.url,
        };
    }
    return {
        type: 'github',
        href: ctaButton.url,
    };
}
function updateFont(config) {
    const font = config.font;
    const result = {};
    if (!font)
        return undefined;
    if ('family' in font) {
        return transformFontDetails(font);
    }
    function transformFontDetails(details) {
        const { url } = details, rest = __rest(details, ["url"]);
        return Object.assign(Object.assign({}, rest), (url ? { source: url } : {}));
    }
    if (font.headings) {
        result.heading = transformFontDetails(font.headings);
    }
    if (font.body) {
        result.body = transformFontDetails(font.body);
    }
    return result;
}
function updateFooterSocials(config) {
    var _a, _b, _c;
    if (!((_a = config.footer) === null || _a === void 0 ? void 0 : _a.socials) && !config.footerSocials)
        return undefined;
    const footerSocials = (_c = (_b = config.footer) === null || _b === void 0 ? void 0 : _b.socials) !== null && _c !== void 0 ? _c : config.footerSocials;
    if (Array.isArray(footerSocials)) {
        return footerSocials.reduce((acc, social) => {
            acc[social.type] = social.url;
            return acc;
        }, {});
    }
    return footerSocials;
}
export function upgradeToDocsConfig(config) {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z;
    const fonts = updateFont(config);
    const playground = updateApiPlayground(config);
    const rounded = updateRounded(config);
    const eyebrows = updateEyebrows(config);
    const codeblocks = updateCodeblocks(config);
    const v2Config = Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({ $schema: 'https://mintlify.com/docs.json', theme: 'mint', name: config.name, colors: {
            primary: config.colors.primary,
            light: config.colors.light,
            dark: config.colors.dark,
        }, favicon: config.favicon, navigation: updateNavigationToDocsConfig(config) }, (config.rounded || config.eyebrow || config.codeBlock
        ? {
            styling: {
                rounded,
                eyebrows,
                codeblocks,
            },
        }
        : {})), { logo: config.logo }), (playground ||
        ((_c = (_b = (_a = config.api) === null || _a === void 0 ? void 0 : _a.request) === null || _b === void 0 ? void 0 : _b.example) === null || _c === void 0 ? void 0 : _c.languages) ||
        ((_d = config.api) === null || _d === void 0 ? void 0 : _d.baseUrl) ||
        ((_f = (_e = config.api) === null || _e === void 0 ? void 0 : _e.auth) === null || _f === void 0 ? void 0 : _f.method) ||
        config.openapi
        ? {
            api: Object.assign(Object.assign(Object.assign(Object.assign({}, (config.openapi ? { openapi: config.openapi } : {})), (playground
                ? Object.assign({ display: playground.display }, (playground.proxy !== undefined ? { proxy: playground.proxy } : {})) : {})), (((_j = (_h = (_g = config.api) === null || _g === void 0 ? void 0 : _g.request) === null || _h === void 0 ? void 0 : _h.example) === null || _j === void 0 ? void 0 : _j.languages)
                ? {
                    examples: {
                        languages: config.api.request.example.languages,
                    },
                }
                : {})), (((_k = config.api) === null || _k === void 0 ? void 0 : _k.baseUrl) || ((_m = (_l = config.api) === null || _l === void 0 ? void 0 : _l.auth) === null || _m === void 0 ? void 0 : _m.method)
                ? {
                    mdx: {
                        base: config.api.baseUrl,
                        auth: (_o = config.api.auth) === null || _o === void 0 ? void 0 : _o.method,
                    },
                }
                : {})),
        }
        : {})), (config.modeToggle
        ? {
            appearance: {
                default: (_p = config.modeToggle.default) !== null && _p !== void 0 ? _p : 'system',
                strict: (_q = config.modeToggle.isHidden) !== null && _q !== void 0 ? _q : false,
            },
        }
        : {})), (config.background || config.backgroundImage || config.colors.background
        ? {
            background: Object.assign(Object.assign(Object.assign({}, (config.backgroundImage ? { image: config.backgroundImage } : {})), { decoration: (_r = config.background) === null || _r === void 0 ? void 0 : _r.style }), (config.colors.background
                ? {
                    color: {
                        light: config.colors.background.light,
                        dark: config.colors.background.dark,
                    },
                }
                : {})),
        }
        : {})), (config.topbarLinks
        ? {
            topbar: {
                links: config.topbarLinks.map((item) => {
                    if (item.type === 'link' || item.type === undefined) {
                        return {
                            label: item.name,
                            href: item.url,
                        };
                    }
                    else {
                        return {
                            label: _.capitalize(item.type),
                            href: item.url,
                        };
                    }
                }),
                primary: updateTopbarPrimary(config),
            },
        }
        : {})), (((_s = config.search) === null || _s === void 0 ? void 0 : _s.prompt)
        ? {
            search: {
                prompt: config.search.prompt,
            },
        }
        : {})), (config.metadata || ((_t = config.seo) === null || _t === void 0 ? void 0 : _t.indexHiddenPages) !== undefined
        ? {
            seo: Object.assign(Object.assign({}, (config.metadata ? { metatags: config.metadata } : {})), { indexing: ((_u = config.seo) === null || _u === void 0 ? void 0 : _u.indexHiddenPages) ? 'navigable' : 'all' }),
        }
        : {})), (config.footer || config.footerSocials
        ? {
            footer: Object.assign(Object.assign({}, (((_v = config.footer) === null || _v === void 0 ? void 0 : _v.socials) || config.footerSocials
                ? { socials: updateFooterSocials(config) }
                : {})), (((_w = config.footer) === null || _w === void 0 ? void 0 : _w.links)
                ? {
                    links: config.footer.links.map((group) => {
                        return {
                            header: group.title,
                            items: group.links.map((link) => {
                                return {
                                    label: link.label,
                                    href: link.url,
                                };
                            }),
                        };
                    }),
                }
                : {})),
        }
        : {})), (config.integrations || config.analytics
        ? {
            integrations: Object.assign(Object.assign(Object.assign(Object.assign({}, config.analytics), (((_x = config.integrations) === null || _x === void 0 ? void 0 : _x.intercom)
                ? {
                    intercom: {
                        appId: config.integrations.intercom,
                    },
                }
                : {})), (((_y = config.integrations) === null || _y === void 0 ? void 0 : _y.frontchat)
                ? {
                    frontchat: {
                        snippetId: config.integrations.frontchat,
                    },
                }
                : {})), (((_z = config.integrations) === null || _z === void 0 ? void 0 : _z.osano)
                ? {
                    osano: {
                        scriptSource: config.integrations.osano,
                    },
                }
                : {})),
        }
        : {})), (fonts ? { fonts } : {})), (config.redirects ? { redirects: config.redirects } : {}));
    return v2Config;
}
